File: CS/GoogleCalendarExample/Form1.cs
using Google.Apis.Auth.OAuth2;
using Google.Apis.Calendar.v3;
using Google.Apis.Calendar.v3.Data;
using Google.Apis.Services;
using Google.Apis.Util.Store;
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace GoogleCalendarExample {
    public partial class Form1 : Form {
        static string[] Scopes = { CalendarService.Scope.Calendar };
        static string ApplicationName = "GoogleCalendarExample";
        
        public Form1() {
            InitializeComponent();
            UpdateFormState();
            schedulerControl1.Start = DateTime.Now;
        }

        CalendarService CalendarService { get; set; }
        CalendarListEntry CalendarEntry { get; set; }

        void Synchronize() {
            this.dxGoogleCalendarSync1.CalendarService = CalendarService;
            this.dxGoogleCalendarSync1.CalendarId = CalendarEntry.Id;
            this.dxGoogleCalendarSync1.Synchronize();
        }

        void OnBtnConnectClick(object sender, EventArgs e) {
            cbCalendars.SelectedValueChanged -= OnCbCalendarsSelectedValueChanged;
            UserCredential credential;
            using(var stream =
                new FileStream("secret\\client_secret.json", FileMode.Open, FileAccess.Read)) {
                string credPath = System.Environment.GetFolderPath(
                    System.Environment.SpecialFolder.Personal);
                credPath = Path.Combine(credPath, ".credentialss");

                credential = GoogleWebAuthorizationBroker.AuthorizeAsync(
                    GoogleClientSecrets.Load(stream).Secrets,
                    Scopes,
                    "user",
                    CancellationToken.None,
                    new FileDataStore(credPath, true)).Result;
                Log("Credential file saved to: " + credPath);
            }

            // Create Google Calendar API service.
            CalendarService = new CalendarService(new BaseClientService.Initializer() {
                HttpClientInitializer = credential,
                ApplicationName = ApplicationName,
            });
            var calendarListRequtes = CalendarService.CalendarList.List();
            CalendarList calendarList = calendarListRequtes.Execute();
            foreach(var item in calendarList.Items) {
                Log(item.Summary);
            }
            cbCalendars.DisplayMember = "Summary";
            cbCalendars.DataSource = calendarList.Items;
            cbCalendars.SelectedValueChanged += OnCbCalendarsSelectedValueChanged;
            CalendarEntry = this.cbCalendars.SelectedValue as CalendarListEntry;
            Synchronize();
            UpdateFormState();
        }
        void OnCbCalendarsSelectedValueChanged(object sender, EventArgs e) {
            CalendarEntry = this.cbCalendars.SelectedValue as CalendarListEntry;
            this.dxGoogleCalendarSync1.Storage = null;
            this.schedulerStorage1.Appointments.Clear();
            this.dxGoogleCalendarSync1.Storage = this.schedulerStorage1;
            Synchronize();            
            UpdateFormState();
        }

        void UpdateFormState() {
            if(CalendarService == null) {
                this.cbCalendars.Enabled = false;
                this.btnUpdate.Enabled = false;
                this.btnConnect.Enabled = true;
            }
            else {
                this.cbCalendars.Enabled = true;
                this.btnUpdate.Enabled = true;
                this.btnConnect.Enabled = false;
            }
        }

        void Log(string message) {
            this.tbLog.Text += message + "\r\n";
        }
        void Log(string format, params object[] args) {
            Log(string.Format(format, args));
        }
        void SetStatus(string message) {
            this.tsStatus.Text = message;
        }

        void OnBtnUpdateClick(object sender, EventArgs e) {
            Synchronize();
        }       
    }
}

File: VB/GoogleCalendarExample/Form1.vb
Imports Microsoft.VisualBasic
Imports Google.Apis.Auth.OAuth2
Imports Google.Apis.Calendar.v3
Imports Google.Apis.Calendar.v3.Data
Imports Google.Apis.Services
Imports Google.Apis.Util.Store
Imports System
Imports System.Collections.Generic
Imports System.IO
Imports System.Linq
Imports System.Text
Imports System.Threading
Imports System.Threading.Tasks
Imports System.Windows.Forms

Namespace GoogleCalendarExample
	Partial Public Class Form1
		Inherits Form
		Private Const DefaultCaption As String = "Google Calendar Importer"
		Private Shared Scopes() As String = { CalendarService.Scope.CalendarReadonly }
		Private Shared ApplicationName As String = "GoogleCalendarExample"

		Public Sub New()
			InitializeComponent()
			UpdateFormState()
		End Sub

		Private privateCalendarService As CalendarService
		Private Property CalendarService() As CalendarService
			Get
				Return privateCalendarService
			End Get
			Set(ByVal value As CalendarService)
				privateCalendarService = value
			End Set
		End Property

		Private Sub OnBtnConnectClick(ByVal sender As Object, ByVal e As EventArgs) Handles btnConnect.Click, btnUpdate.Click
			Dim credential As UserCredential
			Using stream = New FileStream("secret\client_secret.json", FileMode.Open, FileAccess.Read)
				Dim credPath As String = System.Environment.GetFolderPath(System.Environment.SpecialFolder.Personal)
				credPath = Path.Combine(credPath, ".credentials")

				credential = GoogleWebAuthorizationBroker.AuthorizeAsync(GoogleClientSecrets.Load(stream).Secrets, Scopes, "user", CancellationToken.None, New FileDataStore(credPath, True)).Result
				Log("Credential file saved to: " & credPath)
			End Using

			' Create Google Calendar API service.
			CalendarService = New CalendarService(New BaseClientService.Initializer() With {.HttpClientInitializer = credential, .ApplicationName = ApplicationName})
			Dim calendarListRequtes = CalendarService.CalendarList.List()
			Dim calendarList As CalendarList = calendarListRequtes.Execute()
			For Each item In calendarList.Items
				Log(item.Summary)
			Next item
			cbCalendars.DisplayMember = "Summary"
			cbCalendars.DataSource = calendarList.Items

			AddHandler cbCalendars.SelectedValueChanged, AddressOf OnCbCalendarsSelectedValueChanged
			UpdateFormState()
		End Sub
		Private Sub OnCbCalendarsSelectedValueChanged(ByVal sender As Object, ByVal e As EventArgs)
			Dim calendarEntry As CalendarListEntry = TryCast(Me.cbCalendars.SelectedValue, CalendarListEntry)
			Dim calendar As Calendar = CalendarService.Calendars.Get(calendarEntry.Id).Execute()
			Dim events As Events = CalendarService.Events.List(calendarEntry.Id).Execute()
			Log("Loaded {0} events", events.Items.Count)
			Me.schedulerStorage1.Appointments.Items.Clear()
			Me.schedulerStorage1.BeginUpdate()
			Try
				Dim importer As New CalendarImporter(Me.schedulerStorage1)
				importer.Import(events.Items)
			Finally
				Me.schedulerStorage1.EndUpdate()
			End Try
			SetStatus(String.Format("Loaded {0} events", events.Items.Count))
			UpdateFormState()
		End Sub

		Private Sub UpdateFormState()
			If CalendarService Is Nothing Then
				Me.cbCalendars.Enabled = False
				Me.btnUpdate.Enabled = False
				Me.btnConnect.Enabled = True
			Else
				Me.cbCalendars.Enabled = True
				Me.btnUpdate.Enabled = True
				Me.btnConnect.Enabled = False
			End If
		End Sub

		Private Sub Log(ByVal message As String)
			Me.tbLog.Text += message & Constants.vbCrLf
		End Sub
		Private Sub Log(ByVal format As String, ParamArray ByVal args() As Object)
			Log(String.Format(format, args))
		End Sub
		Private Sub SetStatus(ByVal message As String)
			Me.tsStatus.Text = message
		End Sub
	End Class
End Namespace
